backends/p4tools/modules/testgen/main.cpp

```diff
@@ -0,0 +1,34 @@
+#include "lib/crash.h"
+#include "lib/exceptions.h"
+
+#include "backends/p4tools/modules/testgen/testgen.h"
+
+int main(int argc, char** argv) {
+    setup_signals();
+
+    std::vector<const char*> args;
+    for (int i = 0; i < argc; ++i) {
+        args.push_back(argv[i]);
+    }
+
+    try {
+        P4Tools::P4Testgen::Testgen().main(args);
+    } catch (const Util::CompilerBug& e) {
```

Comment:
We should add a util function and replace `CompilerBug` to `ToolsBug`, `TestgenBug` or something (not in this PR).